### PR TITLE
Login component and authorisation

### PR DESCRIPTION
**Pull request by wiktorks**
Added login UI component and user authorisation module.

### CODE REVIEW

File: client/src/Authorization/Login/Login.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+class Login extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            username: "",
+            password: "",
+            submitted: false, 
+            redirect: false
+        };
+        if(authService.authenticateUser() === null) { // If user already logged in, we redirect him to main page
```
Comment (SzymonSmykala) [inline client/src/Authorization/Login/Login.js:16]: remove comments and create function: redirectToMainPage

File: client/src/Authorization/Login/Login.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        this.setState({ submitted: true });
+        if (!(username && password)) return;
+
+        const response = await authService.login(username, password);
+        if(response.success) {
+            const user = { // object containing user authorisation data
+                name: username,
+                token: response.token
+            }
+            window.localStorage.setItem('user', JSON.stringify(user)); // user data kept in local storage under 'user' key
```
Comment (SzymonSmykala) [inline client/src/Authorization/Login/Login.js:34]: just remove comments everywhere. 
'user' key should be stored in a static class.

File: client/src/Authorization/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
+import React, { Component } from "react";
+import { Redirect } from "react-router-dom";
+import { authService } from "../_auth-services/authentication";
+import "./Login.css";
+import bookImage from "../../pictures/books.png";
+
+class Login extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
```
Comment (SzymonSmykala) [inline client/src/Authorization/Login/Login.js:10]: As far as I understand we will use the user entity (Dto class) in the whole application. Shouldn't we have a user entity that can be retrieved from local storage? 

Like this one:
user = { 
                 name: username,
                 token: response.token
       }

**Comment by SzymonSmykala** [comment]
Where are unit tests?

File: client/src/Authorization/Login/Login.js (hunk 3 above)
Comment (wiktorks) [reply client/src/Authorization/Login/Login.js:10]: First this object has to be created and added to localStorage. It is added in user login. If the user logs out, this object is removed.

File: client/src/Authorization/Login/Login.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        if (!(username && password)) return;
+
+        const response = await authService.login(username, password);
+        if(response.success) {
+            const user = {
+                name: username,
+                token: response.token
+            }
+            authService.authenticateUser(user);
+            this.redirectToMainPage();
```
Comment (ZPoreba) [inline client/src/Authorization/Login/Login.js:43]: After successful sign in you can redirect to url /signedin and add empty route in routes.js ( e.g. <Route path='/signedin' />)

File: client/src/Authorization/Login/Login.js (hunk 3 above)
Comment (SzymonSmykala) [reply client/src/Authorization/Login/Login.js:10]: Not sure if I understand. Nevermind!

**Pull request merged**